src/Type/Equality.purs

```diff
@@ -14,10 +15,22 @@ module Type.Equality
 -- | Note: any instance will necessarily overlap with
 -- | `refl` below, so instances of this class should
 -- | not be defined in libraries.
+class TypeEquals :: forall k. k -> k -> Constraint
 class TypeEquals a b | a -> b, b -> a where
-  to :: a -> b
-  from :: b -> a
+  typeEqualsProof :: forall p. p a -> p b
```

Comment:
Maybe just `proof`?

Reply:
I think `proof` is more likely to be used as a local variable name, which is why I was thinking of giving this a longer name. I guess it depends whether we expect this module to be imported qualified or not?

Reply:
I was imagining it'll be qualified - that's also the pattern we've been encouraging in most places - dropping type/class names from functions and members, like all the `Ref` stuff a while back. 

With it already having names like `to` and `from` I've only ever imported it qualified as it is.

Reply:
Yeah, true. It seems Type.Prelude currently re-exports the entirety of this module unqualified, which I think is perhaps not ideal. Come to think of it I'm not sure if people tend to import Type.Prelude qualified or not. Maybe we could just not re-export the new `proof` from Type.Prelude?

Reply:
Not re-exporting `proof` sounds fine to me :+1:.

I'm not sure I've ever imported `Type.Prelude` at all, but if I were to do so I'd probably import it qualified, but then I import almost everything qualified these days, unless it's in our app prelude.

Reply:
Type.Prelude re-exports only TypeEquals, from and to so nothing to change there then o/